File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.Templated/MessageItem.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI

Public Class MessageItem
	Inherits Control
	Implements INamingContainer
	Private parentControl As Message
	Public Sub New(ByVal parent As Message)
		parentControl = parent
	End Sub

	Public ReadOnly Property Text() As String
		Get
			Return parentControl.Text
		End Get
	End Property
End Class

File: aspnet-in-practice/06/CS/CustomControls.Templated/MessageItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;

namespace CustomControls.Templated
{
	public class MessageItem : Control, INamingContainer
	{
		private Message parentControl;
		public MessageItem(Message parent)
		{
			parentControl = parent;
		}

		public string Text
		{
			get
			{
				return parentControl.Text;
			}
		}
	}
}